resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/LocalDirsFeatureStep.scala

```diff
@@ -66,10 +76,19 @@ private[spark] class LocalDirsFeatureStep(
     val containerWithLocalDirVolumeMounts = new ContainerBuilder(pod.container)
       .addNewEnv()
         .withName("SPARK_LOCAL_DIRS")
-        .withValue(resolvedLocalDirs.mkString(","))
+        .withValue(localDirs.mkString(","))
         .endEnv()
       .addToVolumeMounts(localDirVolumeMounts: _*)
       .build()
     SparkPod(podWithLocalDirVolumes, containerWithLocalDirVolumeMounts)
   }
+
+  def findLocalDirVolumeMount(pod: SparkPod): Seq[String] = {
+    val localDirVolumes = pod.pod.getSpec.getVolumes.asScala
```

Comment:
Hmm... do you need to list the volumes at all? Can't you just look at the mounts (since they already have the volume name)?